### PR TITLE
React context - Lyn Kim

### PR DESCRIPTION
**Pull request by lyn-kim**

### CODE REVIEW

File: react-context/flow.txt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  set to false which renders the page with the Auth React component where the
+  user has the option to either sign-in or sign-up
+- If the path chosen is to sign-up, the sign-up page is rendered
+- Once user provides a new username and password then click 'Register', client's
+  POST request is made to api/auth/sign-up where:
+    - the server first checks to see if the request body contains both username
+      and password (along with its values)
+          - If not, a ClientError is thrown
+          - If both username and password are provided:
+            - the login info is stringified and saved in this.state
```
Comment (thebearingedge) [inline react-context/flow.txt:14]: this part is out of place, as the request has already been sent to the server.

File: react-context/flow.txt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  has the option to either sign-in or sign-up
+- If the path chosen is to sign-in, the sign-in page is rendered
+- Once user provides new username and password and click 'Log In':
+    - the user's token gets stored in localStorage for future use
+    - client's POST request is sent to api/auth/sign-in where:
+      - the server checks to see if the request body contains both the username
+        and password (along with its values)
+        - If not, a ClientError is thrown
+        - If both username and password are provided:
+          - login info is stringified and saved in this.state
```
Comment (thebearingedge) [inline react-context/flow.txt:36]: this part is out of place, as the request has already been sent to the server.

**Review by thebearingedge — APPROVED** [review]
Ok, very good, but there are a couple of weird spots.

File: react-context/flow.txt (hunk 1 above)
Comment (lyn-kim) [reply react-context/flow.txt:14]: line 14 moved to be right after "Once user provides a new username and password, then click 'Register'"

File: react-context/flow.txt (hunk 2 above)
Comment (lyn-kim) [reply react-context/flow.txt:36]: line 36 moved to be right after "Once user provides a new username and password, then click 'Log-in'"

**Pull request merged**